README.md

```diff
@@ -4,37 +4,105 @@
+
+Once the message is on the `Inbound MESH Queue` the Message Translator component picks each one up and performs following steps:
+
+1) Initial verification
+
+    This step parses the EDIFACT message and verifies general structure of the message and can result in two types of errors:
+    - Interchange Header (UNB) is missing mandatory fields (sender, recipient, sequence number, NHSACK request flag).
+
+        Since the header can't be parsed, an NHSACK can't be produced to inform the laboratory about the error. This is a critical error and manual intervention is required. Message is rejected causing it to be placed back on the `Inbound MESH Queue`. It is picked up again for retry until the MQ max retry value is reached. After that, message is placed on the Dead Letter Queue and manual intervention is required.
+    - Interchange Trailer (UNZ) message count or Message Header (UNH) or Message Trailer (UNT) can't be parsed.
+
+        Since the Interchange Header was parsed, all the information required to produce an NHSACK is available. If Interchange Header has the NHSACK request flag set, an NHSACK is produced and sent back to the laboratory MESH mailbox.
+
+2) Translation
 
-The GP System will receive medical reports on the `Outbound GP Queue` in form of a FHIR DSTU3 Bundle.
+    At this step EDIFACT has already beebn parsed and split into messages. Each message is translated into FHIR Bundle. Any error occurring at this step, is logged and if NHSACK is requested, a proper information is set in it. Because of the error, no FHIR Bundle is created for that single message. Other messages are not affected and can still be processed. At the end of this step, a summary NHSACK is generated containing information about all the successes and failures of EDIFACT messages processing.
+
+3) Sending to `Outbound GP Queue`
+
+    All individual EDIFACT messages that have been successfully translated into FHIR are sent to `Outbound GP Queue`. If an error occurrs at this step, whole EDIFACT is rejected and it returns to the `Inbound MESH Queue` from where it will be picked up again until MQ max retry value is reached. After that, message is placed on the Dead Letter Queue and manual intervention is required.
+
+4) Sending NHSACK to `Outbound MESH Queue`
+
+    All individual FHIR messages have been sent to the `Outbound GP Queue`. An NHSACK has been generated and is being sent to `Outbound MESH Queue`. If an error occurrs at this step, whole EDIFACT is rejected and it returns to the `Inbound MESH Queue` from where it will be picked up again until MQ max retry value is reached. After that, message is placed on the Dead Letter Queue and manual intervention is required. Since the message has been already put on `Inbound MESH Queue`, it will be processed again causing duplicates to appear on the `Outbound GP Queue`. Handling duplicates is been described in the [duplicates](#duplicates) section.
```

Comment:
```suggestion
    All individual FHIR messages have been sent to the `Outbound GP Queue`. An NHSACK has been generated and is being sent to `Outbound MESH Queue`. If an error occurs at this step, the whole EDIFACT is rejected, and it returns to the `Inbound MESH Queue` from where it will be picked up again until MQ max retry value is reached. After that, message is placed on the Dead Letter Queue and manual intervention is required. Since the message has been already put on `Inbound MESH Queue`, it will be processed again causing duplicates to appear on the `Outbound GP Queue`. Handling duplicates has been described in the [duplicates](#duplicates) section.
```